test/test_nn.py

```diff
@@ -1299,6 +1311,17 @@ def check():
         modules.clear()
         check()
 
+        # verify the correct exception is thrown when calling ModuleDict
+        try:
```

Comment:
nit: is there a reason you are not using `self.assertRaises`?

Reply:
I am afraid I am not familiar with all the `self.assert*`

Reply:
You can grep for example usage in other tests

Reply:
fixed. Thank you for guiding me through this, and making this PR, albeit tiny, of a high quality.